pkg/cvo/cvo.go

```diff
@@ -611,12 +612,50 @@ func versionString(update configv1.Update) string {
 	return "<unknown>"
 }
 
-// currentVersion returns an update object describing the current known cluster version.
-func (optr *Operator) currentVersion() configv1.Update {
-	return configv1.Update{
-		Version: optr.releaseVersion,
-		Image:   optr.releaseImage,
+// currentVersion returns an update object describing the current
+// known cluster version.  Values from the upstream Cincinnati service
+// are used as fallbacks for any properties not defined in the release
+// image itself.
+func (optr *Operator) currentVersion() configv1.Release {
+	return optr.mergeReleaseMetadata(optr.release)
+}
+
+// mergeReleaseMetadata returns a deep copy of the input release.
+// Values from any matching availableUpdates release are used as
+// fallbacks for any properties not defined in the input release.
+func (optr *Operator) mergeReleaseMetadata(release configv1.Release) configv1.Release {
```

Comment:
I would have named this as `updateReleaseMetadata()` but we can handle this in another PR.

Reply:
It doesn't update in place, it returns a new object merging data from two sources